### PR TITLE
Fixes #120

### LINKED ISSUE #120: Settings link is only valid if the user is the currently logged in user

### PR DESCRIPTION
**Pull request by pmuir**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: src/app/header/header.component.ts [hunk 1]
```diff
@@ -9,17 +10,29 @@ import { Context } from './../models/context';
...
   providers: []
 })
-
 export class HeaderComponent implements OnInit {
   title = 'Almighty';
   imgLoaded: Boolean = false;
 
+  menuCallbacks = new Map<String, MenuHiddenCallback>([
+    [
+      'settings', function (context, menuItem) {
```
Comment (joshuawilson) [inline src/app/header/header.component.ts]: menuItem is not used.

File: src/app/header/header.component.ts [hunk 2]
```diff
@@ -9,17 +10,29 @@ import { Context } from './../models/context';
...
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
   providers: []
 })
-
 export class HeaderComponent implements OnInit {
   title = 'Almighty';
   imgLoaded: Boolean = false;
 
+  menuCallbacks = new Map<String, MenuHiddenCallback>([
```
Comment (joshuawilson) [inline src/app/header/header.component.ts]: For flow sake, I think functions should be below the constructor. It's not a big deal.

File: src/app/models/menu-item.ts [hunk 3]
```diff
@@ -1,8 +1,12 @@
-export class MenuItem {
+import { Context } from './context';
```
Comment (joshuawilson) [inline src/app/models/menu-item.ts]: Why are these imported? You are not using them.

**Comment by joshuawilson** [comment]
This doesn't appear to fix the problem in #120. If I'm logged out, I can't see http://localhost:3000/pmuir/settings. There are errors.

File: src/app/header/header.component.ts (hunk 1 above)
Comment (pmuir) [inline src/app/header/header.component.ts]: Will remove.

File: src/app/header/header.component.ts (hunk 2 above)
Comment (pmuir) [inline src/app/header/header.component.ts]: Declaring variables after the constructor is a linting error.

File: src/app/models/menu-item.ts (hunk 3 above)
Comment (pmuir) [inline src/app/models/menu-item.ts]: Good spot.

**Comment by pmuir** [comment]
Ok, commit updated with changes requested.

Good catch on the logged out point - I hadn't thought through that scenario - I added a couple of guards to neatly handle that.

File: src/app/header/header.component.ts (hunk 2 above)
Comment (joshuawilson) [inline src/app/header/header.component.ts]: yes, you are right. I must have looked at it too fast and thought it was a function instead.

**Pull request merged**